pkg/ingester/ingester.go

```diff
@@ -805,7 +810,12 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, pushReq *push.Request) (
 			level.Warn(i.logger).Log("msg", "failed to rollback appender on error", "user", userID, "err", err)
 		}
 
-		return nil, err
+		// Check for a particular per-instance limit and return that error directly
+		// since it contains extra information for gRPC and our logging middleware.
+		if errors.Is(err, errMaxInMemorySeriesReached) {
+			return nil, err
+		}
+		return nil, wrapWithUser(err, userID)
```

Comment:
I'm not sure about the implications of wrapping every error that could be returned by `pushSamplesToAppender()`.

Reply:
From what I could tell, the only cases where `pushSamplesToAppender()` was directly returning errors, it was already calling `wrapWithUser()` for the errors.

Reply:
You're right.